webview-ui/src/utils/command-validation.ts

```diff
@@ -127,18 +127,49 @@ export function containsDangerousSubstitution(source: string): boolean {
  * chaining operators (&&, ||, ;, |, or &) and newlines.
  *
  * Uses shell-quote to properly handle:
- * - Quoted strings (preserves quotes)
+ * - Quoted strings (preserves quotes and newlines within quotes)
  * - Subshell commands ($(cmd), `cmd`, <(cmd), >(cmd))
  * - PowerShell redirections (2>&1)
  * - Chain operators (&&, ||, ;, |, &)
- * - Newlines as command separators
+ * - Newlines as command separators (but not within quotes)
  */
 export function parseCommand(command: string): string[] {
 	if (!command?.trim()) return []
 
-	// Split by newlines first (handle different line ending formats)
+	// First, protect newlines inside quoted strings by replacing them with a placeholder
+	// This prevents splitting multi-line quoted strings (e.g., git commit -m "multi\nline")
+	const quotedStringPlaceholder = "___NEWLINE_IN_QUOTE___"
+	let protectedCommand = command
+
+	// Track quote state and replace newlines inside quotes
+	let inDoubleQuote = false
+	let inSingleQuote = false
+	let result = ""
+
+	for (let i = 0; i < command.length; i++) {
```

Comment:
this probably still is to fragile, for instance:

`echo "test foo"'"'"bar"`

echoes:

`test foo"bar`

and this algorithm will think bar isnt within quotes. We probably at least need to improve this or maybe use some library